hack/jenkins/README.md

```diff
@@ -41,6 +41,39 @@ outside this repository, it's tricky to keep documentation for it up
 to date quickly. However, the scripts themselves attempt to provide
 color for the configuration(s) that each script runs in.
 
+## GCS Log Format
+
+Our `upload-to-gcs.sh` script runs at the start and end of every job. Logs on
+post-commit Jenkins go under `gs://kubernetes-jenkins/logs/`. Logs on PR
+Jenkins go under `gs://kubernetes-jenkins-pull/pr-logs/pull/PULL_NUMBER/`.
+Individual run logs go into the `JOB_NAME/BUILD_NUMBER` folder.
+
+At the start of the job, it uploads `started.json` containing the version and
+timestamp.
+
+At the end, it uploads `finished.json` containing the result and timestamp, as
+well as the build log into `build-log.txt`. Under `artifacts/` we put our
+test results in `junit_XY.xml`, along with gcp resource lists and cluster logs.
+
+It also updates `latest-build.txt` at the end to point to this build number.
+In the end, the directory structure looks like this:
+
+```
+gs://kubernetes-jenkins/logs/kubernetes-e2e-gce/
+  latest-build.txt
+  12345/
+    build-log.txt
+    started.json
+    finished.json
+    artifacts/
+      gcp-resources-{before, after}.txt
+      junit_{00, 01, ...}.xml
+      jenkins-e2e-master/{kube-apiserver.log, ...}
+      jenkins-e2e-minion-abcd/{kubelet.log, ...}
+  12344/
```

Comment:
:+1: might be worth calling out which of these are required for munger to do its thing?